obspy/io/shapefile/core.py

```diff
@@ -0,0 +1,282 @@
+# -*- coding: utf-8 -*-
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+from future.utils import native_str
+
+from obspy import Catalog
+from obspy.core.event import Origin, Magnitude
+from obspy.core.inventory import Inventory
+
+try:
+    import gdal
+    from osgeo import ogr, osr
+except ImportError as e:
+    has_GDAL = False
+    IMPORTERROR_MSG = str(e) + (
+        ". ObsPy's write support for shapefiles requires the 'gdal' module "
+        "to be installed in addition to the general ObsPy dependencies.")
+else:
+    has_GDAL = True
+    gdal.UseExceptions()
+
+
+def _write_shapefile(obj, filename, **kwargs):
+    """
+    Write :class:`~obspy.core.event.Catalog` object to a ESRI shapefile.
+
+    :type obj: :class:`~obspy.core.event.Catalog` or
+        :class:`~obspy.core.inventory.Inventory`
+    :param obj: ObsPy object for shapefile output
+    :type filename: str or file
+    :param filename: Filename to write to. According to ESRI shapefile
+        definition, multiple files with the following suffixes will be written:
+        ".shp", ".shx", ".dbj", ".prj".
+    """
+    if not has_GDAL:
+        raise ImportError(IMPORTERROR_MSG)
+    if not filename.endswith(".shp"):
+        filename += ".shp"
+
+    driver = ogr.GetDriverByName(native_str("ESRI Shapefile"))
+
+    driver.DeleteDataSource(filename)
+    data_source = driver.CreateDataSource(filename)
+
+    # create the layer
+    if isinstance(obj, Catalog):
+        _add_catalog_layer(data_source, obj)
+    elif isinstance(obj, Inventory):
+        _add_inventory_layer(data_source, obj)
+    else:
+        msg = ("Object for shapefile output must be a Catalog or Inventory.")
+        raise TypeError(msg)
+
+
+def _add_catalog_layer(data_source, catalog):
+    """
+    :type data_source: :class:`osgeo.ogr.DataSource`.
+    :param data_source: OGR data source the layer is added to.
+    :type catalog: :class:`~obspy.core.event.Catalog`
+    :param catalog: Event data to add as a new layer.
+    """
+    if not has_GDAL:
+        raise ImportError(IMPORTERROR_MSG)
+
+    # [name, type, width, precision]
+    # field name is 10 chars max
+    # ESRI shapefile attributes are stored in dbf files, which can not
+    # store datetimes, only dates, see:
+    # http://www.gdal.org/drv_shapefile.html
+    # use POSIX timestamp for exact origin time, set time of first pick
+    # for events with no origin
+    field_definitions = [
+        ["EventID", ogr.OFTString, 100, None],
+        ["OriginID", ogr.OFTString, 100, None],
+        ["Magnitu_ID", ogr.OFTString, 100, None],
```

Comment:
I don't care..

It's 10 char max and the automagic routines are stripping long names to 8 chars with underscore/numeric counters. But of course I didn't want to rip apart the "ID" part and thought I'd rather use what length is available to make it as clear as possible what this is referring to. But I guess I might as well change it..